src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -269,22 +270,33 @@ CHIP_ERROR Instance::Read(const ConcreteReadAttributePath & aPath, AttributeValu
     case Attributes::FeatureMap::Id:
         return aEncoder.Encode(mFeatureFlags);
 
+    case Attributes::ClusterRevision::Id:
+        return aEncoder.Encode(kCurrentClusterRevision);
+
     case Attributes::SupportedWiFiBands::Id:
-        VerifyOrReturnError(mFeatureFlags.Has(Feature::kWiFiNetworkInterface), CHIP_NO_ERROR);
-        VerifyOrReturnError(mpDriver.Valid(), CHIP_NO_ERROR);
+#if (!CHIP_DEVICE_CONFIG_ENABLE_WIFI_STATION && !CHIP_DEVICE_CONFIG_ENABLE_WIFI_AP)
+        return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+#else
+        VerifyOrReturnError(mFeatureFlags.Has(Feature::kWiFiNetworkInterface), CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute));
 
-        return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR {
-            WiFiBand bandsBuffer[kMaxWiFiBands];
-            Span<WiFiBand> bands(bandsBuffer);
-            ReturnErrorOnFailure(mpDriver.Get<WiFiDriver *>()->GetSupportedWiFiBands(bands));
+        return aEncoder.EncodeList([this](const auto & encoder) {
+            uint32_t bands = mpDriver.Get<WiFiDriver *>()->GetSupportedWiFiBands();
+            static_assert(chip::to_underlying(WiFiBandEnum::kUnknownEnumValue) <= std::numeric_limits<uint32_t>::digits,
+                          "Expected WiFiBandEnum::kUnknownEnumValue to fit in uint32_t's number of bits");
 
-            for (WiFiBand band : bands)
+            // Extract every band from the bitmap of supported bands, starting positionally on the right.
+            for (uint32_t band_bit_pos = 0; band_bit_pos < chip::to_underlying(WiFiBandEnum::kUnknownEnumValue); ++band_bit_pos)
```

Comment:
Nothing really guarantees, in general, that the "unknown" value is larger than all the known values....  This is making some assumptions.

Might be simpler to just go up to the number of bits that can be in uint32_t or something.

Reply:
Will do.